common/src/main/java/io/netty/util/internal/logging/InternalLoggerFactory.java

```diff
@@ -85,12 +85,14 @@ private static InternalLoggerFactory useLog4J2LoggerFactory(String name) {
 
     private static InternalLoggerFactory useLog4JLoggerFactory(String name) {
         try {
+            // Fail immediately if the Log4j Priority class is not on the class path.
+            // This works around a Graal issue with build-time class initialisation.
+            Class.forName("org.apache.log4j.Priority");
+
             InternalLoggerFactory f = Log4JLoggerFactory.INSTANCE;
             f.newInstance(name).debug("Using Log4J as the default logging framework");
             return f;
-        } catch (LinkageError ignore) {
-            return null;
-        } catch (Exception ignore) {
+        } catch (Throwable ignore) {
             // We catch Exception and not ReflectiveOperationException as we still support java 6
```

Comment:
Hmm is this really what we want to do as this also would make this comment not correct. Why not just also catch ClassNotFoundExeption?